src/base/bittorrent/tracker.h

```diff
@@ -48,54 +53,58 @@ namespace BitTorrent
 {
     struct Peer
     {
-        QHostAddress ip;
         QByteArray peerId;
-        int port;
+        QHostAddress address;  // reported by socket
+        QByteArray claimedAddress;  // self claimed by peer
```

Comment:
I wouldn't store these two addresses here. The one above is only needed once so it can be stored in AnnounceRequest directly. Here you need to store the resulting address only (either claimed or socket). It also can be stored as std::string to prevent it from conversion each time you create response.

Reply:
>I wouldn't store these two addresses here. The one above is only needed once so it can be stored in AnnounceRequest directly. Here you need to store the resulting address only (either claimed or socket). It 

I don't see how to do that while preserving the current behavior.

Reply:
>while preserving the current behavior.

What part of current behavior do you mean?

Reply:
I need it here for each recorded peer:
https://github.com/qbittorrent/qBittorrent/blob/aef6a21ec194f3c1a2613891dcd987e2c2de77d4/src/base/bittorrent/tracker.cpp#L420

Reply:
>Here you need to store the resulting address only (either claimed or socket). 

I think merging the 2 addresses is not a good idea. Both are different things and merging it together throws away information that might be useful later when debugging.

Reply:
On a second thought, I should definitely precompute and cache the `ip` response field, so it is changed.